versioned/storage/store/src/test/java/org/projectnessie/versioned/storage/versionstore/TestVersionStoreImpl.java

```diff
@@ -156,4 +165,80 @@ public Reference updateReferencePointer(
         fromMessage("commit foo"),
         singletonList(Put.of(ContentKey.of("some-key"), IcebergTable.of("meta", 42, 43, 44, 45))));
   }
+
+  @ParameterizedTest
+  @MethodSource
+  public void commitWithDatabaseTimeout(
+      @SuppressWarnings("unused") String name,
+      BiFunction<AtomicBoolean, Persist, Persist> delegateFactory)
+      throws Exception {
+    VersionStore store = new VersionStoreImpl(persist);
+
+    BranchName branch = BranchName.of("branch1");
+    Hash branch1 = store.create(branch, Optional.empty()).getHash();
+
+    AtomicBoolean called = new AtomicBoolean();
+    Persist tested = delegateFactory.apply(called, persist);
+
+    VersionStore storeTested = new VersionStoreImpl(tested);
+    storeTested.commit(
+        branch,
+        Optional.of(branch1),
+        fromMessage("commit foo"),
+        singletonList(Put.of(ContentKey.of("some-key"), IcebergTable.of("meta", 42, 43, 44, 45))));
+
+    soft.assertThat(called).isTrue();
+  }
+
+  public static Stream<Arguments> commitWithDatabaseTimeout() {
+    return Stream.of(
+        arguments(
+            "updateReferencePointer",
+            (BiFunction<AtomicBoolean, Persist, Persist>)
+                (called, p) ->
+                    new PersistDelegate(p) {
+                      @Nonnull
+                      @Override
+                      public Reference updateReferencePointer(
+                          @Nonnull Reference reference, @Nonnull ObjId newPointer)
+                          throws RefNotFoundException, RefConditionFailedException {
+                        if (called.compareAndSet(false, true)) {
+                          throw new UnknownOperationResultException("test", new RuntimeException());
```

Comment:
Could we also simulate persisted writes (`updateReferencePointer`) with a timeout (`UnknownOperationResultException`)?

Reply:
Sorry - but, have you read the body of `public static Stream<Arguments> commitWithDatabaseTimeout()`?

Reply:
I have, but I do not see where it simulates a timeout after a persisted write... I might have missed it, though.

Reply:
(see above - and apologies that I sounded too harsh)

Reply:
no worries :+1: